docs/pages/includes/role-spec.mdx

```diff
@@ -58,26 +58,35 @@ spec:
     max_connections: 2
     # limit number of concurrent Kubernetes sessions per user
     max_kubernetes_connections: 1
-    # Define how Teleport deals with session recording failures, such as a full
-    # disk error. The value can be set to either `best_effort` or `strict`. If
-    # set to `strict`, the session will terminate immediately. If set to
-    # `best_effort`, the session won’t be terminated, and the recording will be
-    # disabled. The configuration is done per service (currently, only `ssh` is
-    # supported).
+    # Specify whether or not to record the user's SSH sessions and how to deal
+    # with session recording failures, such as a full disk error.
+    #
+    # Session recording is enabled if any of the user's roles specify
+    # `best_effort` or `strict`, where, in case of recording failures, the
+    # session will be immediately terminated only if the mode is set to `strict`.
+    #
+    # The configuration is done per service. The `default` will be used if the
+    # service doesn’t specify a value. When set in multiple roles, the value
+    # used will always be the strictest, following the rule: `strict` > `best_effort` > `off`.
+    # To turn off the session recording, all user’s roles must set the value to
+    # `off`; otherwise, the strictest value will be used.
     record_session:
       # Specify whether or not to record the user's desktop sessions.
       # Desktop session recording is enabled if one or more of the user's
-      # roles has enabled recording. Defaults to true if unspecified.
+      # roles has enabled recording. Supported values are `true` (enables
+      # recording) and `false` (turns off recording). Defaults to `true` if
+      # unspecified.
+      #
       # Desktop sessions will never be recorded if auth_service.session_recording
       # is set to 'off' in teleport.yaml (Auth Service) or if the cluster's
       # session_recording_config resource has set 'mode: off'.
       desktop: true
       # Optional: the default session recording mode to use when a
-      # protocol-specific mode is not set.
-      default: best_effort|strict
-      # Optional: Session recording mode for SSH sessions (Teleport Server
-      # Access). If not set, the value set on default will be used.
-      ssh: best_effort|strict
+      # protocol-specific mode is not set. Defaults to `best_effort`
+      default: off|best_effort|strict
```

Comment:
What confuses me is the `default` section. desktop does not follow SSH and `best_effort` and `strict` may not apply to other services. 

I would imagine each service would want its own set of "modes":
```
record_session:
  desktop: true # true | false
  ssh: off # off | best_effort | strict
  db: full # full | queries_only
```